libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m

```diff
@@ -594,10 +592,9 @@ + (SFSDKOAuthClient *)idpAuthInstance:(SFSDKOAuthClientConfig *)config {
     } else {
         instance = [[SFSDKIDPAuthClient alloc] initWithConfig:config];
     }
-    SFSDKMutableOAuthClientContext *context = [[SFSDKMutableOAuthClientContext alloc] initWithAuthType:SFOAuthTypeAdvancedBrowser];
+    SFSDKMutableOAuthClientContext *context = [[SFSDKMutableOAuthClientContext alloc] initWithAuthType:SFOAuthTypeIDP];
```

Comment:
This was the underlying cause of most of the hybrid remote refresh problems we were seeing specific to `iOS`, incorrectly assigning `OAuthClientContext`, most likely caused by a copy-paste error, and it was quite hard to hunt down. Let me explain:
- Going through a lot of async magical steps, we get to a point where `SFSDKOAuthClient` spins up an instance of itself with some context based on some input.
- There were some class methods for this (1 of each type) - `idpAuthInstance`, `nativeBrowserAuthInstance`, and `webviewAuthInstance`.
- Unfortunately, all these methods initialized themselves with the same auth type, i.e., `SFOAuthTypeAdvancedBrowser`, regardless of what the input was. :-(
- I fixed this by assigning the correct type to each.
- The next issue was there was no way to create a client with auth type `SFOAuthTypeRefresh`. I added a utility method for this and added a check in the creation block to use this auth type if a valid refresh token is present.
- The hybrid view controller relies on the auth type to assign the value of `createAbsUrl` (if it's of type `SFOAuthTypeRefresh`, it sets it to `NO`, which is what we want in this case). Since the type was incorrectly being set to `SFOAuthTypeAdvancedBrowser`, the downstream refresh flow was failing because this would lead to incorrect computation of `retURL`. I reverted the other special case since we don't need it anymore after fixing the underlying issue @khawkins.
cc: @wmathurin @trooper2013

Reply:
Nice sleuthing, @bhariharan !  I think @trooper2013 owes you a beer. 😉